### PR TITLE
Custom fields

### LINKED ISSUE #16: Allow users to create a custom field
Right now all parameters are hard coded and there can be a brand/social network which might not be in the options. It will be useful if an option to add a custom field is available

### PR DESCRIPTION
**Pull request by Riktam-Santra**
## Description

Added a customSocials field which allows users to create their own fields with their parameters including icons, name and url

Resolves #16 

## Basic Checklist

- [x] ✨ Ran `dartdoc`
- [x] 🛠️ Added proper documentation
- [x] ❌ No warnings or errors in the code
- [x] ✅ If added a new parameter, added in Readme and example too

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by Riktam-Santra** [comment]
In the example I have added my socials since I didn't have a good example, you might want to change that so I haven't updated the screenshot yet on README

**Comment by Riktam-Santra** [comment]
~~Also, looks like your repository has been marked as excluded, you might want to enquire about that~~
Looks like they removed exculsion on your repository now

File: README.md [hunk 1]
```diff
@@ -51,6 +51,12 @@ class MyApp extends StatelessWidget {
           tagLine: 'Flutter Developer',
           twitterHandle: 'AbhishekDoshi26',
           instagramUserName: '_abhishek_doshi',
+          customSocials: [
+            CustomSocialField(
+                icon: const Icon(Icons.abc),
+                name: "karna.satva",
+                url: "https://discordapp.com/users/712546092612583424"),
```
Comment (AbhishekDoshi26) [inline README.md]: Use fake names instead of real ones.

File: example/lib/main.dart [hunk 2]
```diff
@@ -37,6 +38,12 @@ class MyApp extends StatelessWidget {
           instagram: '_abhishek_doshi',
           facebookHandle: '_abhishek_doshi',
           tiktokUrl: "https://www.tiktok.com/@AbhishekDoshi26",
+          customSocials: [
+            CustomSocialField(
+                icon: const Icon(Icons.abc),
+                name: "karna.satva",
+                url: "https://discordapp.com/users/712546092612583424"),
```
Comment (AbhishekDoshi26) [inline example/lib/main.dart]: Same here

File: lib/models/custom_field.dart [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/material.dart';
+
+class CustomSocialField {
+  final Icon icon;
+  final String name;
+  final String url;
+
+  CustomSocialField(
+      {required this.icon, required this.name, required this.url});
```
Comment (AbhishekDoshi26) [inline lib/models/custom_field.dart]: Add a trailing comma.

**Review by AbhishekDoshi26 — CHANGES_REQUESTED** [review]
Also, inside models folder, there should be a barrlel file that will export cutom_field.dart
Same way, inside contactus.dart, add an export to models/models.dart

File: lib/models/custom_field.dart (hunk 3 above)
Comment (Riktam-Santra) [reply lib/models/custom_field.dart]: I committed the requested changes

Comment (Riktam-Santra) [reply lib/models/custom_field.dart]: @AbhishekDoshi26

**Review by AbhishekDoshi26 — APPROVED** [review]

**Pull request merged**